js/product/productController.js

```diff
@@ -0,0 +1,100 @@
+firmApp.controller('productCtrl', function ($scope, $http, $uibModal, $filter, localModelProduct, localRepositoryProduct) {
+
+    $scope.productList = [];
+
+    localRepositoryProduct.getDataProduct (function(data){
+        $scope.productList = data;
+    });
```

Comment:
bardzo ladnie to wyglada - jak widzisz teraz w kontrzulerze masz tylko proste wywolanie funkcji odczytu danych, ktora jest w innej klasie. Kazda klasa ma "tematyczny" podzial i ma jeden, konkretny sens. Dzieki temu o wiele latwiej sie czyta kod, i wszystko jest od razu jasne.